rust/xaynet-server/src/storage/impls.rs

```diff
@@ -334,3 +334,18 @@ pub enum SumDictDeleteError {
     #[error("sum participant does not exist")]
     DoesNotExist,
 }
+
+#[macro_export]
+macro_rules! retry {
+    ($function: expr, $duration: expr) => {
+        loop {
+            match $function.await {
```

Comment:
Good point. I took the second option because the connection is normally consumed after a method call. It still works  in the macro because it is an expression but I think that could be confusing as well.